RecoEgamma/EgammaHLTProducers/plugins/EgammaHLTHcalVarProducerFromRecHit.cc

```diff
@@ -0,0 +1,233 @@
+      effectiveAreas_(config.getParameter<std::vector<double> >("effectiveAreas")),
+      absEtaLowEdges_(config.getParameter<std::vector<double> >("absEtaLowEdges")),
+      recoEcalCandidateProducer_(consumes(config.getParameter<edm::InputTag>("recoEcalCandidateProducer"))),
+      hbheRecHitsTag_(consumes(config.getParameter<edm::InputTag>("hbheRecHitsTag"))),
+      rhoProducer_(doRhoCorrection_ ? consumes<double>(config.getParameter<edm::InputTag>("rhoProducer"))
+                                    : edm::EDGetTokenT<double>()),
+      caloGeometryToken_{esConsumes()},
+      hcalTopologyToken_{esConsumes()},
+      hcalChannelQualityToken_{esConsumes(edm::ESInputTag("", "withTopo"))},
+      hcalSevLvlComputerToken_{esConsumes()},
+      caloTowerConstituentsMapToken_{esConsumes()},
+      putToken_{produces<reco::RecoEcalCandidateIsolationMap>()} {
+  if (doRhoCorrection_) {
+    if (absEtaLowEdges_.size() != effectiveAreas_.size()) {
+      throw cms::Exception("IncompatibleVects") << "absEtaLowEdges and effectiveAreas should be of the same size. \n";
+    }
+
+    if (absEtaLowEdges_.at(0) != 0.0) {
+      throw cms::Exception("IncompleteCoverage") << "absEtaLowEdges should start from 0. \n";
+    }
+
+    for (unsigned int aIt = 0; aIt < absEtaLowEdges_.size() - 1; aIt++) {
+      if (!(absEtaLowEdges_.at(aIt) < absEtaLowEdges_.at(aIt + 1))) {
+        throw cms::Exception("ImproperBinning") << "absEtaLowEdges entries should be in increasing order. \n";
+      }
+    }
+  }
+}
+
+void EgammaHLTHcalVarProducerFromRecHit::fillDescriptions(edm::ConfigurationDescriptions &descriptions) {
+  edm::ParameterSetDescription desc;
+
+  desc.add<edm::InputTag>(("recoEcalCandidateProducer"), edm::InputTag("hltRecoEcalCandidate"));
+  desc.add<edm::InputTag>(("rhoProducer"), edm::InputTag("fixedGridRhoFastjetAllCalo"));
+  desc.add<edm::InputTag>(("hbheRecHitsTag"), edm::InputTag("hltHbhereco"));
+  desc.add<bool>(("doRhoCorrection"), false);
+  desc.add<double>(("rhoMax"), 999999.);
+  desc.add<double>(("rhoScale"), 1.0);
+  //eThresHB/HE are from RecoParticleFlow/PFClusterProducer/python/particleFlowRecHitHBHE_cfi.py
+  desc.add<std::vector<double> >(("eThresHB"), {0.1, 0.2, 0.3, 0.3});
+  desc.add<std::vector<double> >(("etThresHB"), {0, 0, 0, 0});
+  desc.add<std::vector<double> >(("eThresHE"), {0.1, 0.2, 0.2, 0.2, 0.2, 0.2, 0.2});
+  desc.add<std::vector<double> >(("etThresHE"), {0, 0, 0, 0, 0, 0, 0});
+  desc.add<double>(("innerCone"), 0);
+  desc.add<double>(("outerCone"), 0.14);
+  desc.add<int>(("depth"), 0);
+  desc.add<int>(("maxSeverityHB"), 9);
+  desc.add<int>(("maxSeverityHE"), 9);
+  desc.add<bool>(("doEtSum"), false);
+  desc.add<bool>(("useSingleTower"), false);
+  desc.add<std::vector<double> >("effectiveAreas", {0.079, 0.25});  // 2016 post-ichep sinEle default
+  desc.add<std::vector<double> >("absEtaLowEdges", {0.0, 1.479});   // Barrel, Endcap
+  descriptions.add(("hltEgammaHLTHcalVarProducerFromRecHit"), desc);
```

Comment:
Why is there an extra bracket-pair around the first argument?

Reply:
Ah I just copied from other .cc files in that folder.. but you are right that these extra braces are not needed. 
Let me remove them and update the PR. Thanks!